tools/bin/commands/release

```diff
@@ -235,6 +234,105 @@ update_image_versions() {
     sed "s/TagMajor: .*$/TagMajor: \"$next_version\"/" -i $topdir/install/operator/build/conf/config.yaml
 }
 
+get_github_username() {
+    if [ -z "${GITHUB_USERNAME:-}" ]; then
+        echo "ERROR: environment variable GITHUB_USERNAME has not been set."
+        echo "Please populate it with your github id"
+        return 1
+    fi
+    echo $GITHUB_USERNAME
+}
+
+get_github_access_token() {
+    if [ -z "${GITHUB_ACCESS_TOKEN:-}" ]; then
+        echo "ERROR: environment variable GITHUB_ACCESS_TOKEN has not been set."
+        echo "Please populate it with a valid personal access token from github (with 'repo', 'admin:org_hook' and 'admin:repo_hook' scopes)."
+        return 1
+    fi
+    echo $GITHUB_ACCESS_TOKEN
+}
+
+prepare_binaries() {
+    local from=$1
+    local to=$2
+
+    if ! [[ -d ${from} ]]; then
+        echo "ERROR: The directory where the binaries are located must be a valid directory, got [${from}]"
+        return 1
+    fi
+
+    if ! [[ -d ${to} ]]; then
+        mkdir ${to}
+    fi
+
+    for dist in darwin-amd64 linux-amd64 windows-amd64; do
+        cp ${from}/${dist}/syndesis-operator ${to}/syndesis-operator-${dist}
+    done
+}
+
+get_gh_release_name() {
+    local tag=$1
+    local current_branch=$(git branch  | grep \* | awk '{print $2}')
+    local name=${tag}
+
+    if [[ "${current_branch}" == "master" ]]; then
+        name="${tag}-${current_branch}"
+    fi
+
+    echo ${name}
+}
+
+get_gh_release_desc() {
+    local desc="Please follow the instructions at http://syndesis.io/quickstart/ to install syndesis."
+    local commit_history=""
+
+    previous_release_commit=$(curl -s --fail https://api.github.com/repos/syndesisio/syndesis/releases/latest | \
+        grep  target_commitish | \
+        cut -d '"' -f 4)
+
+    if [[ -z $previous_release_commit ]]; then
+        echo "WARNING: Unable to fetch previous release commit"
+    else
+        commit_history=$(git log --pretty=format:"%h - %s" ${previous_release_commit}..HEAD)
+    fi
+
+    desc="${desc}\n\n$commit_history"
+
+    echo ${desc}
+}
+
+publish_artifacts() {
+    local release_dir=$1
+    local tag=$2
+
+    local release_name=$(get_gh_release_name ${tag})
+    local desc=$(get_gh_release_desc)
+
+    get_github_access_token
+    get_github_username
+
+    set +e
+    local upload_url=$(curl -q --fail -X POST -u $GITHUB_USERNAME:${GITHUB_ACCESS_TOKEN} -H "Accept: application/vnd.github.v3+json" -H "Content-Type: application/json" -d "{\"tag_name\": \"${tag}\", \"name\": \"${release_name}\", \"body\": \"${desc}\"}" https://api.github.com/repos/syndesisio/syndesis/releases | jq -r .upload_url | cut -d{ -f1)
```

Comment:
```suggestion
    local data="{
	  \"tag_name\": \"${tag}\",
	  \"name\": \"${release_name}\",
	  \"body\": \"${desc}\"}
	"
	local upload_url=$(curl -q --fail \
      -X POST \
      -u $GITHUB_USERNAME:$GITHUB_ACCESS_TOKEN \
      -H "Accept: application/vnd.github.v3+json" \
      -H "Content-Type: application/json" \
      -d "$data" \
      https://api.github.com/repos/syndesisio/syndesis/releases | jq -r '.upload_url | sub("{.*"; "")'
```

The payload `-d` bit could be a nice variable and parameters to curl could be on individual lines, to help with readability, `jq` can handle search-replace of strings.